tests/test_reference_processing.py

```diff
@@ -0,0 +1,45 @@
+import logging
+from bs4 import BeautifulSoup  # type: ignore
+from jupyter_book_to_htmlbook.reference_processing import process_interal_refs
+
+
+class TestInternalRefs:
```

Comment:
Put all these in a test class, assuming that we'll throw the bibliography reference tests in this file as well.